### PR TITLE
docs: listing past versions

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

listing past versions with cross-link to relevant docs version
Note: referring to RTFD but it shall be directing to lightning docs when we populate them

https://pytorch-lightning--17014.org.readthedocs.build/en/17014/past_versions.html

image

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda

### CODE REVIEW

File: docs/source-pytorch/past_versions.rst [hunk 1]
```diff
@@ -0,0 +1,127 @@
+Past PyTorch lightning versions
```
Comment (lantiga) [inline docs/source-pytorch/past_versions.rst:1]:
```suggestion
Past PyTorch Lightning versions
```

File: docs/source-pytorch/past_versions.rst [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+===============================
+
+The Pytorch Lightning evolved over time so to keep track, let us share brief version history list with links to particular docs of the given version.
```
Comment (lantiga) [inline docs/source-pytorch/past_versions.rst:4]:
```suggestion
Pytorch Lightning evolved over time. Here's the full history of PyTorch Lightning versions with links to the respective docs.
```

File: docs/source-pytorch/past_versions.rst [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+
+.. list-table:: Past versions
+   :widths: 10 50 40
+   :header-rows: 1
+
+   * - Version
+     - Title
+     - Particular Bug-fix versions
+
+   * - `1.9 <https://github.com/Lightning-AI/lightning/releases/tag/1.9.0>`_
```
Comment (lantiga) [inline docs/source-pytorch/past_versions.rst:15]: we need to list them from last (e.g. 1.9.3) at the top, to first (1.9.0) at the bottom, within each minor

File: docs/source-pytorch/past_versions.rst [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+The Pytorch Lightning evolved over time so to keep track, let us share brief version history list with links to particular docs of the given version.
+
+
+.. list-table:: Past versions
+   :widths: 10 50 40
+   :header-rows: 1
+
+   * - Version
+     - Title
+     - Particular Bug-fix versions
```
Comment (lantiga) [inline docs/source-pytorch/past_versions.rst:13]:
```suggestion
     - Bug-fix versions
```

**Review by lantiga — CHANGES_REQUESTED** [review]
A few nits

File: docs/source-pytorch/past_versions.rst (hunk 3 above)
Comment (Borda) [reply docs/source-pytorch/past_versions.rst:15]: These are in single line

**Review by lantiga — APPROVED** [review]

File: docs/source-pytorch/past_versions.rst [hunk 5]
```diff
@@ -0,0 +1,127 @@
+Past PyTorch Lightning versions
+===============================
+
+Pytorch Lightning evolved over time. Here's the full history of PyTorch Lightning versions with links to the respective docs.
```
Comment (carmocca) [inline docs/source-pytorch/past_versions.rst:4]:
```suggestion
PyTorch Lightning evolved over time. Here's the complete history of versions with links to their respective docs.
```

**Review by carmocca — APPROVED** [review]
Looks great

**Pull request merged**